### PR TITLE
Fixed overflow bug when loading large PLY files

### PR DESCRIPTION
**Pull request by pierotofy**
This problem was partially fixed in https://github.com/PointCloudLibrary/pcl/pull/2161, but a few places remained where the overflow still occurred, causing the point cloud to be initialized to all zeros.

cc @dakotabenjamin

### CODE REVIEW

**Review by smathermather — APPROVED** [review]
We should also push this fix back up to the upstream project as well.

**Comment by pierotofy** [comment]
On its way! https://github.com/PointCloudLibrary/pcl/pull/2715

**Comment by pierotofy** [comment]
I don't have merge permissions on this repo so either @smathermather or @dakotabenjamin should merge this in (after review is completed). :+1:

**Pull request merged**
